### DISCUSSION TITLE
Is it possible to delete all data scopes in a connection with few clicks on UI?

### DISCUSSION THREAD

**Discussion by Stepness** [apache/devlake#7872] [Q&A]
Currently Id like to delete a connection, but Im blocked because I have to delete all associated data scopes first.
![](https://github.com/user-attachments/assets/bc1b2844-4002-40a4-8cef-58d70af4a363)

Is there a button to delete them all in one go? Because I can't find it
I saw in previous discussion that an implementation to delete single scopes has been done in #5558

[a comment by an automated account is not shown]

**Comment by klesh** [comment]
Not that I know of. All scope configs of the connection must be deleted one by one first.
Maybe you could write a script to delete them in one go.

**Reply by Stepness** [reply]
It looks very impractical. Can this feature become a To do?

**Reply by klesh** [reply]
I am not sure, it is a rare and risky operation, and it is hard to say if it should be available on the UI.
@Startrekzky What do you think?

**Reply by Startrekzky** [reply]
I also think it's a risky operation.

**Reply by limberger** [reply]
it can be risky, but now I for example need to delete more than 1000 scopes ...
